frontend/src/static/js/v_authorship.js

```diff
@@ -103,9 +106,36 @@ window.vAuthorship = {
       this.totalLineCount = totalLineCount;
       res.sort((a, b) => b.lineCount - a.lineCount);
 
+      this.filesLineInfo = this.convertLineCountToString(filesLineInfoArray);
       this.files = res;
       this.isLoaded = true;
     },
+
+    addLineCountToFileType(path, lineCount, filesLineInfoArray) {
+      const out = {};
+      var fileType = path.split(".").pop();
+      out.fileType = (fileType.length === 0) ? "others" : fileType;
+      out.lineCount = lineCount;
+      var index = filesLineInfoArray.findIndex((x) => x.fileType === out.fileType);
+      if (index === -1) {
+        filesLineInfoArray.push(out);
+      } else {
+        filesLineInfoArray[index].lineCount += out.lineCount;
+      }
+    },
+
+    convertLineCountToString(filesLineInfoArray) {
+      var filesLineInfo = "";
+      for (var key in filesLineInfoArray) {
+        if (Object.prototype.hasOwnProperty.call(filesLineInfoArray, key)) {
+            filesLineInfo += filesLineInfoArray[key].fileType + ": "
+            + filesLineInfoArray[key].lineCount + ", ";
+        }
+      }
+
+      filesLineInfo = filesLineInfo.substring(0, filesLineInfo.lastIndexOf(", "));
```

Comment:
instead of this, how about we use an array and use .join method? would be cleaner and also more efficient.

Reply:
@eugenepeh can u elaborate? do u mean use ```.join()``` method only for line 136?